### DISCUSSION TITLE
Issues in the Grafana dashboard

### DISCUSSION THREAD

**Discussion by askrinnik** [nazarii-piontko/sample-dotnet-otel#3] [General]
Hi Nazarii,

I tried to use the updated version of [the dashboard](https://grafana.com/grafana/dashboards/17706-asp-net-otel-metrics/) and found it doesn't work.
Also, I noticed that when I imported the panel, there was no possibility to select my Prometheus data source.
I figured out that the dashboard uses the hardcoded Prometheus uid. It happens when the dashboard is exported **without** the '**Export for sharing externally**' option.
![image](https://github.com/nazarii-piontko/sample-dotnet-otel/assets/1359187/7990b1db-cb30-4d55-a26b-fdf7cf8a23e7)

To confirm my conclusion, I did the following:
I loaded your repository and started containers. The dashboard works fine here. 
I exported the Grafana dashboard **with the external sharing option** and imported it to my Grafana instance. It became work.

So, it seems to me it is necessary to redeploy the dashboard to the Grafana library with the external sharing option.

Best regards,
Oleksandr

**Comment by askrinnik** [comment]
Also, I found some metrics, which were not renamed. `_ms_` is still used instead of `_milliseconds_`.
![image](https://github.com/nazarii-piontko/sample-dotnet-otel/assets/1359187/4e1b9d9a-02c0-458f-a904-fb5a446b88b1)

**Reply by nazarii-piontko** [reply]
Hi Oleksandr, thank you for your observation, I will fix it, just a little bit later due to a real lack of time, sorry for the delay.

**Comment by askrinnik** [comment]
> Hi Oleksandr, thank you for your observation, I will fix it, just a little bit later due to a real lack of time, sorry for the delay.



> Hi Oleksandr, thank you for your observation, I will fix it, just a little bit later due to a real lack of time, sorry for the delay.

No problem, I fixed that locally. Moreover, I rewrote your dashboard for metrics, scrapped from OTEL Collector. :)

BTW, I didn't find how to publish own dashboard in the Grafana Gallery. Is it payed feature?

**Comment by nazarii-piontko** [comment]
Fixed, thank you @askrinnik for your valuable contribution :+1:
